### PR TITLE
Enable ellipse method for specifying a region

### PR DESCRIPTION
**Pull request by MiCurry**
This PR enables the ability to specify elliptical regions using an ellipse pts specification method.

### CODE REVIEW

**Comment by MiCurry** [comment]
@mgduda I just tested creating regions over both of the poles, and I didn't seem to have a problem. Was it a problem when changing the orientation of an ellipse that's on a pole that was having problems?

File: limited_area/region_spec.py [hunk 1]
```diff
@@ -22,6 +22,14 @@ def create_bound_method(func, obj):
...
 
 
+def normal(lon):
+    if lon > np.pi:
+        return lon - 2.0 * np.pi
+    elif lon < -np.pi:
+        return lon + 2.0 * np.pi
+    else:
+        return lon
+
```
Comment (mgduda) [inline limited_area/region_spec.py]: I think we can delete the `normal` function -- I had been using this for debugging, but forgot to remove it.

File: limited_area/region_spec.py [hunk 2]
```diff
@@ -165,3 +184,62 @@ def circle(self, center_lat, center_lon, radius):
...
+        # Convert ellipse center from (lat,lon) to Cartesian
+        C = latlon_to_xyz(center_lat, center_lon, 1.0)
+
+        # Convert semi-major and semi-minor axis lengths from meters to radians
+        semi_major = semi_major / EARTH_RADIUS
+        semi_minor = semi_minor / EARTH_RADIUS
+
+        a = semi_major
+        b = semi_minor
```
Comment (mgduda) [inline limited_area/region_spec.py]: I think we can safely remove `a` and `b`.

File: limited_area/region_spec.py [hunk 3]
```diff
@@ -165,3 +184,62 @@ def circle(self, center_lat, center_lon, radius):
...
+            K[2] = 0
+
+        # S is then a vector orthogonal to C
+        S = np.cross(C, K)
+        S = S / np.linalg.norm(S)
+
+        for r in np.linspace(0.0, 2.0*np.pi, 100):
+            radius = np.sqrt((semi_major * np.cos(r))**2 + (semi_minor * np.sin(r))**2)
+            P0 = rotate_about_vector(C, S, radius)
+            d = rotate_about_vector(P0, C, r)
```
Comment (mgduda) [inline limited_area/region_spec.py]: It looks like I also forgot to delete `d` -- it was presumably just used for debugging purposes.

**Comment by MiCurry** [comment]
@mgduda Thanks for the look over. I resolved the issue you had within your comments. I also added a new commit that contains the Japan ellipse example.

**Review by mgduda — APPROVED** [review]
This looks fine to me.

**Pull request merged**